pkg/tsdb/influxdb/models/query.go

```diff
@@ -44,6 +95,96 @@ func (query *Query) Build(queryContext *backend.QueryDataRequest) (string, error
 	return res, nil
 }
 
+func (query *Query) renderAdhocFilters() []string {
+	res := make([]string, 0, len(query.AdhocFilters))
+	for i, tag := range query.AdhocFilters {
+		str := ""
+
+		if i > 0 {
+			if tag.Condition == "" {
+				str += "AND"
+			} else {
+				str += tag.Condition
+			}
+			str += " "
+		}
+
+		// If the operator is missing we fall back to sensible defaults
+		if tag.Operator == "" {
+			if regexpOperatorPattern.MatchString(tag.Value) {
+				tag.Operator = "=~"
+			} else {
+				tag.Operator = "="
+			}
+		}
+
+		isOperatorTypeHandler := func(tag *Tag) (string, string) {
+			// Attempt to identify the type of the supplied value
+			var lowerValue = strings.ToLower(tag.Value)
+			var r = regexp.MustCompile(`^(-?)[0-9\.]+$`)
```

Comment:
idk if this is worth calling out. feel free to ignore
```suggestion
			lowerValue := strings.ToLower(tag.Value)
			r := regexp.MustCompile(`^(-?)[0-9\.]+$`)
```